### PR TITLE
PR to fix Checkov Security alerts

### PR DESCRIPTION
**Pull request by patrickmukumbu**
# Description

This PR fixes the following checkov alerts and also uses S3 module in organization example:

- fix: CKV_AWS_33: "Ensure KMS key policy does not contain wildcard (*) principal"
- fix: CKV_AWS_109: Ensure IAM policies does not allow permissions management / resource exposure without constraints
- fix: CKV_AWS_144: Ensure that S3 bucket has cross-region replication enabled
- fix: CKV_AWS_18: Ensure the S3 bucket has access logging enabled
- fix: CKV_AWS_145: Ensure that S3 buckets are encrypted with KMS by default. NOTE:: Resource for this feature not currently detected by checkov though buckets are encrypted.
- feat: Used upgraded S3 module for external bucket example (organization trail).

## Features/Fixes/Patches/Changes list:
Please check the [CHANGELOG.md](https://github.com/boldlink/terraform-aws-cloudtrail/blob/fix/security-fixes/CHANGELOG.md#123---2023-02-07)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [ ] OWNER: Does your submission pass?
    * [x] Pre-commit (attach logs/print screen to this PR)
    * [x] Checkov/terrascan (attach logs/print screen to this PR)
    * [x] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [ ] First PR? Have you deleted `## How to use this template...` from the root `.README.md`?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -46,7 +48,7 @@ locals {
           Sid    = "Allow principals in the account to decrypt log files"
           Effect = "Allow"
           Principal = {
-            AWS = ["*"]
+            AWS = ["arn:${local.partition}:iam::${local.account_id}:root"] ## changed from '*'
```
Comment (hugoalmeida264) [inline locals.tf]: No need for the comment here and bellow.

**Review by hugoalmeida264 — CHANGES_REQUESTED** [review]
See my comments.

I noted that both kms policies are setup so that only administrators or power users can use the key, there is the possible use case where a role who is neither needs access to the trail, if it is not now please add a note on the changelog to address it.

File: locals.tf (hunk 1 above)
Comment (patrickmukumbu) [reply locals.tf]: @hugoalmeida264  Addressed

**Review by mariomerco — COMMENTED** [review]
Would be nice to have the "complete" example with `var.replication_configuration`

File: examples/complete/locals.tf [hunk 2]
```diff
@@ -86,13 +87,76 @@ locals {
...
         }
   ] })
 
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Action" : "sts:AssumeRole",
+        "Principal" : {
+          "Service" : "s3.amazonaws.com"
```
Comment (hugoalmeida264) [inline examples/complete/locals.tf]: use a data source for amazonaws.com

File: examples/complete/main.tf [hunk 3]
```diff
@@ -1,16 +1,61 @@
...
+}
+
+provider "aws" {
+  alias  = "dest"
+  region = "eu-central-1"
+}
+
+module "replication_role" {
+  source                = "boldlink/iam-role/aws"
+  name                  = "${local.name}-replication-role"
```
Comment (hugoalmeida264) [inline examples/complete/main.tf:13]: Version missing.

File: examples/complete/main.tf [hunk 4]
```diff
@@ -1,16 +1,61 @@
...
+  force_detach_policies = true
+  policies = {
+    "${local.name}-replication-policy" = {
+      policy = local.role_policy
+    }
+  }
+}
+
+module "replication_kms_key" {
+  source           = "boldlink/kms/aws"
```
Comment (hugoalmeida264) [inline examples/complete/main.tf:25]: version missing

File: examples/complete/main.tf [hunk 5]
```diff
@@ -1,16 +1,61 @@
...
+  alias_name       = "alias/${local.replication_bucket}-key-alias"
+  tags             = local.tags
+
+  providers = {
+    aws = aws.dest
+  }
+}
+
+module "replication_bucket" {
+  source                 = "boldlink/s3/aws"
```
Comment (hugoalmeida264) [inline examples/complete/main.tf:38]: Version missing

**Review by hugoalmeida264 — CHANGES_REQUESTED** [review]

**Review by hugoalmeida264 — APPROVED** [review]

**Pull request merged**